### ISSUE TITLE
Config for separate tests

### ISSUE THREAD

**Issue by nvnieuwk** [askimed/nf-test#28]
Hi, thanks again for the quick fix yesterday!

While doing some more testing, I ran into the problem that I couldn't really specify any configuration to a specific test since the `nextflow.config` file specified in `nf-test.config` is applied to all tests. I then tried to supply the configuration inside the `when` clause like this:
```
when {
    process {
         withName: GATK4_COLLECTSVEVIDENCE {
             ext.args = {"--sample-name ${meta.id}"}
         }
    }
}
```

But I got following error when trying this: 
```
  groovy.lang.MissingMethodException: No signature of method: main_nf$_run_closure1$_closure2.GATK4_COLLECTSVEVIDENCE() is applicable for argument types: (main_nf$_run_closure1$_closure2$_closure4$_closure6$_closure7) values: [main_nf$_run_closure1$_closure2$_closure4$_closure6$_closure7@22ee2d0]
```

I've also tried inserting the configuration in the `mock.nf` using the `when: workflow` clause, but that didn't work either since the `mock.nf` isn't a configuration file itself.

So my suggestion is to add a possibility to specify a config file in the tests itself or to have a `config` clause in- or outside of the `when` clause in which you can specify specific configuration for each test.

Cheers,
Nicolas

**Comment by lukfor** [comment]
Hi again,

an inline Nextflow configuration (like in your example) is not an easy fix. But adding a `config` property should be much easier. For example, you could then set it in the testsuite for all tests:

```
nextflow_process {

    name "Test Process..."
    script "main.nf"
    process "my_process"
    config "path/to/test/nextflow.config"

    ...

}
```

Or you could overwrite the `config` property for a specific test:

```
nextflow_process {

   test("my test") {

      config "path/to/test/nextflow.config"
      ...

    }

}
```

This config is than **added** to the default configuration located in `tests/nextflow.config` (e.g. `nextflow -c tests/nextflow.config -c path/to/test/nextflow.config ...`).

What do you think?

**Comment by lukfor** [comment]
> Hi, thanks for the fast reply! I think that would be an even better solution than what I proposed. I'm not sure which of these would be the better choice though. Both could work in our case. Right now we only have one config file for all the tests of module, so a config at the start of the test should be enough. But I can see other people needing a specific config for each test. Would it be a lot of work to implement both?
> 
> Cheers, Nicolas

No, it should be straightforward. I hope to find tomorrow the time to implement it.

Thanks again for your feedback and for taking your time to test it!

**Comment by nvnieuwk** [comment]
Great! Thanks for all the fast responses and flexibility on this!

**Comment by lukfor** [comment]
This feature is now available in version 0.6.2. You could find an example here: https://github.com/askimed/nf-test/blob/main/test-data/process/config/test_process.nf.test

**Comment by nvnieuwk** [comment]
Awesome! Thanks again for the super fast release of this fix! :partying_face:
